### PR TITLE
move vehicle at rest detection ekf2 -> land_detector

### PR DESCRIPTION
**Pull request by dagar**
Obviously there are some differences to consider.
 - centralized vehicle_at_rest detection using current primary IMU instead of per EKF instance
 - filtered angular velocity used for instead of raw IMU delta angle norm
 - is_moving_scaler no longer configurable (EKF2_MOVE_TEST)
 - existing gyro vibration metric was triggering way too easily, the value was increased significantly

I'm happy to make this parameter configurable if anyone cares, but if no one was tuning it I'd rather not have tuning knobs everywhere by default.

Do any of the other changes seem problematic? Technically there's now a circular dependency between the angular velocity (using estimated gyro bias) and EKF vehicle at rest, but it seems fairly harmless in this case.

### CODE REVIEW

File: src/modules/land_detector/LandDetector.cpp [hunk 1]
```diff
@@ -178,4 +196,51 @@ void LandDetector::Run()
...
+			if (!gyro_status_found) {
+				PX4_WARN("IMU status not found for gyro %" PRId32, sensor_selection.gyro_device_id);
+			}
+		}
+	}
+
+	vehicle_imu_status_s imu_status;
+
+	if (_vehicle_imu_status_sub.update(&imu_status)) {
+		static constexpr float GYRO_VIBE_METRIC_MAX = 0.02f; // gyro_vibration_metric * 4.0e4f > is_moving_scaler) TODO: review
```
Comment (bresch) [inline src/modules/land_detector/LandDetector.cpp:233]:
```suggestion
		static constexpr float GYRO_VIBE_METRIC_MAX = 0.02f; // gyro_vibration_metric * dt * 4.0e4f > is_moving_scaler) TODO: review
```

File: src/modules/land_detector/LandDetector.cpp [hunk 2]
```diff
@@ -178,4 +196,51 @@ void LandDetector::Run()
...
+		static constexpr float ACCEL_VIBE_METRIC_MAX = 1.2f; // accel_vibration_metric * dt * 2.1e2f > is_moving_scaler
```
Comment (bresch) [inline src/modules/land_detector/LandDetector.cpp:234]: wasn't the dt = 0.05? So 2.1e-2 / 0.05 =  0.42

Comment (dagar) [reply src/modules/land_detector/LandDetector.cpp:234]: 5 milliseconds = 0.005 (although I had actually used the older value of 4 milliseconds).

Comment (dagar) [reply src/modules/land_detector/LandDetector.cpp:234]: I had to increase the gyro threshold quite a bit before it would stop triggering in SITL.

**Comment by dagar** [comment]
This does screw up the EKF unit tests a bit.

**Review by bresch — DISMISSED** [review]
Looks good to me now

File: src/modules/ekf2/EKF2.cpp [hunk 3]
```diff
@@ -1840,16 +1796,24 @@ void EKF2::UpdateGyroCalibration(const hrt_abstime &timestamp)
 		if (_gyro_cal.last_us != 0) {
 			_gyro_cal.total_time_us += timestamp - _gyro_cal.last_us;
 
-			// consider bias estimates stable when we have accumulated sufficient time
+			// Start checking gyro bias estimates when we have accumulated sufficient calibration time
 			if (_gyro_cal.total_time_us > 30_s) {
+				if (!_gyro_cal.cal_available) {
+					PX4_DEBUG("%d gyro bias now stable", _instance);
```
Comment (dagar) [inline src/modules/ekf2/EKF2.cpp:1802]: Accidentally pulled back in during rebase?

Comment (bresch) [reply src/modules/ekf2/EKF2.cpp:1802]: ouch

**Review by bresch — APPROVED** [review]

**Review by bresch — DISMISSED** [review]

**Comment by dagar** [comment]
Apparently we forgot to merge this. Merging in current master to resolve conflicts.

**Pull request merged**